src/inc/AppxManifestObject.hpp

```diff
@@ -215,8 +215,33 @@ namespace MSIX {
         std::string m_name;
     };
 
+    class AppxManifestOptionalPackageInfo final : public ComClass<AppxManifestOptionalPackageInfo, IAppxManifestOptionalPackageInfo>
+    {
+    public:
+        AppxManifestOptionalPackageInfo(IMsixFactory* factory, std::string mainPackageName) :
+            m_mainPackageName(mainPackageName)
+        {}
+
+        // IAppxManifestOptionalPackageInfo
+        HRESULT STDMETHODCALLTYPE GetIsOptionalPackage(BOOL* isOptionalPackage) noexcept override try
+        {
+            *isOptionalPackage = !m_mainPackageName.empty();

```

Comment:
>*isOptionalPackage = !m_mainPackageName.empty(); [](start = 12, length = 48)

I am not sure, but there might be some issues with converting bool to BOOL

Reply:
Indeed, I would prefer a trinary here to ensure accurate conversion.

---
In reply to: [310279758](https://github.com/Microsoft/msix-packaging/pull/187#discussion_r310279758) [](ancestors = 310279758)